packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -799,53 +804,74 @@ export const sendGsheetsNotification = async (
+            const rows = await projectService.getResultsForChart(
+                user,
+                savedChartUuid,
+            );
 
-            const csvContent = await csv().fromFile(csvUrl.localPath);
             // TODO use csv().fromStream to do stream reading/writting on gsheets
             const refreshToken = await userService.getRefreshToken(
                 scheduler.createdBy,
             );
-            await googleDriveClient.appendToSheet(
-                refreshToken,
-                gdriveId,
-                csvContent,
-            );
+            await googleDriveClient.appendToSheet(refreshToken, gdriveId, rows);
         } else if (dashboardUuid) {
-            if (csvUrls === undefined) {
-                throw new Error('Missing CSV URL');
-            }
+            const dashboard = await dashboardService.getById(
+                user,
+                dashboardUuid,
+            );
+            const chartUuids = dashboard.tiles.reduce<string[]>((acc, tile) => {
+                if (
+                    isDashboardChartTileType(tile) &&
+                    tile.properties.savedChartUuid
+                ) {
+                    return [...acc, tile.properties.savedChartUuid];
+                }
+                return acc;
+            }, []);
 
             const refreshToken = await userService.getRefreshToken(
                 scheduler.createdBy,
             );
-            const googleUploadPromises = csvUrls.map(async (cu) => {
-                const csvContent = await csv().fromFile(cu.localPath);
 
+            // We want to process all charts in sequence, so we don't load all chart results in memory
+            chartUuids.reduce(async (promise, chartUuid) => {
+                await promise;
+                const rows = await projectService.getResultsForChart(
+                    user,
+                    chartUuid,
+                );
+
+                const tile = dashboard.tiles.find(
+                    (t) =>
+                        isDashboardChartTileType(t) &&
+                        t.properties.savedChartUuid === chartUuid,
+                );
+                const chartName =
+                    tile && isDashboardChartTileType(tile)
+                        ? tile.properties.chartName
+                        : undefined;
                 const tabName = await googleDriveClient.createNewTab(
                     refreshToken,
                     gdriveId,
-                    cu.filename,
+                    tile?.properties.title || chartName || chartUuid,
                 );
-                return googleDriveClient.appendToSheet(
+
+                await googleDriveClient.appendToSheet(
                     refreshToken,
                     gdriveId,
-                    csvContent,
+                    rows,
                     tabName,
                 );
-            });
-
-            Promise.all(googleUploadPromises);
+            }, Promise.resolve());
```

Comment:
why do we manually resolve?

Reply:
I think this is how you implement a reduce of async with promises, the `acc` value is a resolved Promise type